packages/sit-onyx/src/utils/comparator.ts

```diff
@@ -0,0 +1,13 @@
+type FlatObject = Record<string, string | boolean | number | null | undefined>;
+
+/**
+ * Compares two flat objects (whose values are only primitives, no arrays or objects)
+ * @returns whether all values match, ignoring in which order they appear
+ */
+export const areObjectsFlatEqual = (obj1: FlatObject, obj2: FlatObject): boolean => {
+  if (Object.keys(obj1).length !== Object.keys(obj2).length) return false;
+  const notEqualItems = Object.entries(obj1).filter(([name, value]) => {
+    return value !== obj2[name];
+  });
```

Comment:
For performance reasons, we don't need to compare all keys. As soo as we have the first mismatch, we can return false

Reply:
I'll change it to .find :)